chore: Convert to monorepo

mayank99 opened the pull request:
Converted this repo into a monorepo using [turborepo](https://turborepo.org/) on top of yarn workspaces. Directory structure is split into `apps/` and `packages/`.
  - Moved all iTwinUI-react files into `packages/`
  - Added a vite playground under `apps/` which uses the linked itwinui-react package just for testing components in a more reliable setting compared to storybook.
  - In the future, the documentation `website` folder will also live under `apps/`

Most of the changes in this PR are simply additions from turborepo/vite and moving files around (git should show those as "renamed").

The only real "changes" are listed below:
- Node 16+ / npm 7+ restriction is now set in root `package.json`.
- iTwinUI-react build was changed to emit files into `cjs`/`esm` folders instead of `lib` so that it works with symlinks. There is no need to copy `package.json` into `lib` anymore.
- Some commands in github workflows had to be updated to reference itwinui-react instead of running from root.
- A lot of jest tests which test for tippy.js visibility started failing so I made some changes to get them to pass.
- Storybook was giving me the following error so I had to add `webpack@5` in the root `package.json`:
  - storybookjs/storybook/issues/15336
- Eslint+prettier was failing so I upgraded some deps and made changes accordingly.
- `ReactNodeArray` is deprecated and was showing an error in `Breadcrumbs` so I changed it to `ReactNode[]`.

A lot of those issues have nothing to do with turborepo but I think they started appearing because dependencies were updated when lockfile was moved.

---

Best way to test these changes:
- Create a fresh clone of this repo and switch to this branch.
- Run `yarn` to install everything.
- Run `yarn dev` to start vite playground and storybook together. 

For running any bespoke commands, you can `cd` into `packages/itwinui-react` or just run it from the repo root by prefixing `yarn workspace @itwin/itwinui-react`.

mayank99 committed:
Move to monorepo setup

mayank99 committed:
Fix build

mayank99 committed:
Oopsie

mayank99 committed:
Fix README

mayank99 committed:
Fix deprecation

mayank99 committed:
Fix publish workflows

mayank99 committed:
Install webpack5 in root

mayank99 committed:
Increase delay in Header tests

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 commented on packages/iTwinUI-react/stories/core/Header.stories.tsx:
```diff
@@ -52,7 +52,7 @@ export default {
   parameters: {
     creevey: {
-      delay: 1000,
+      delay: 2000,
```
The `Full` header visual test is failing. Not sure why because nothing changed. But it's the only failure in the whole repo.
![image](https://user-images.githubusercontent.com/9084735/161339682-94e21c41-ce92-4dd6-ac4c-68ea37b6bd13.png)

I was hoping increasing the delay would fix it but it doesn't. Maybe we just update the image?

veekeys replied:
would be intersting to see, whats the place failing..

mayank99 replied:
Looks like there are some more visual test failures now.
![image](https://user-images.githubusercontent.com/9084735/162478871-ecf81616-83b9-47c0-be6a-e9e0d9c98f0b.png)

veekeys commented on apps/playground/src/main.css:
```diff
@@ -0,0 +1,9 @@
+}
+
+html {
+  font-family: system-ui, sans-serif;
```
shouldnt we take this from css package?

mayank99 replied:
I don't want to add webfont right now (not the point of the PR) but we could do it in a future PR

veekeys replied:
I dont mean adding webfont. I mean defining font family.
It is done with itwinui-css, so you should not need to define anything for demo app.

veekeys commented on apps/playground/src/main.css:
```diff
@@ -0,0 +1,9 @@
+
+html {
+  font-family: system-ui, sans-serif;
+  color-scheme: dark light;
```
does it have influence on our theming?

mayank99 replied:
It influences default colors that the browser provides (e.g. default background, text, scrolllbar, input, etc). I use it in every app I build.

I think we should add it to our theming because we are starting to use system colors in some places (e.g. I noticed `CanvasText` in the breadcrumbs PR).

veekeys replied:
@FlyersPh9 You might want to check it then

veekeys commented on packages/iTwinUI-react/.eslintignore:
```diff
@@ -0,0 +1,13 @@
+**/*.json
```
couldnt move this file, so it shows only as renamed? same with `.eslintrc.js`

mayank99 replied:
not sure what you mean

veekeys replied:
Now it shows that you deleted the one we have and this is a new one. But I assume, this is the same file, just moved.
So it would be better to review when you see actual changes in the file (if there are any) versus seeing this as new file and then manually going to compare with the old one.

mayank99 replied:
Well, I can't change the way git detects a file as changed vs added.

What you could do is use some external diff checker, or even just VSCode (create new folder, run `git init`, create a file, put old contents there, stage the file, then put new contents there, and check diff in the source control). It's not nice but it will show you the changes (there are some).

veekeys commented on packages/iTwinUI-react/.gitignore:
```diff
@@ -0,0 +1,41 @@
+debug.log*
+
+# Compiled Components
+lib/
```
so then we dont have lib anymore ... and same question about multiple gitignore files

veekeys commented on packages/iTwinUI-react/README.md:
```diff
@@ -0,0 +1,99 @@
+<p align="center">
```
hmm.. I would say this still should be in our main readme so users firstly opening github, sees all info.. they dont really care about our monorepo structure or info about it..

mayank99 replied:
I kept this README here for our storybook Overview page and NPM package. I think we need something different for the main README but I agree current info is not enough as it only talks about monorepo

veekeys replied:
Well, i'd say better keep our old readme than showing this. And then we can see, what we want to improve there.
Cause honestly, we still deliver only itwinui-react from here, everything else is applicable only to contributors. So our repo structure should not influence that.

veekeys commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,148 @@
+{
```
could have been better that all old itwinui-react files were shown as moved and new ones created for monorepo...

mayank99 replied:
As I said in the [other comment](https://github.com/iTwin/iTwinUI-react/pull/613#discussion_r846291353), I can't influence the way git detects what's changed vs added.

From what I understand, if there is a new file of the same name as the one that was moved, then the diff will be messed up.

veekeys commented on packages/iTwinUI-react/src/core/Menu/MenuItem.test.tsx:
```diff
@@ -247,25 +245,26 @@ it('should handle key press with sub menus', () => {
   // go right to open sub menu
-  fireEvent.keyDown(menuItem, { key: 'ArrowRight' });
+  menuItem.focus();
+  userEvent.keyboard('{ArrowRight}');
```
so `fireEvent` also stopped working?

veekeys commented on packages/iTwinUI-react/src/core/Table/Table.test.tsx:
```diff
@@ -576,9 +578,6 @@ it('should filter table', () => {
-  const tippy = document.querySelector('[data-tippy-root]') as HTMLElement;
-  expect(tippy.style.visibility).toEqual('hidden');
-
   rows = container.querySelectorAll('.iui-table-body .iui-row');
```
no need to check same `.not.toBeVisible()` ?

mayank99 replied:
I'm checking in `setFilter`

veekeys commented on packages/iTwinUI-react/src/core/Table/Table.test.tsx:
```diff
@@ -666,9 +664,6 @@ it('should not filter table when manualFilters flag is on', () => {
-  const tippy = document.querySelector('[data-tippy-root]') as HTMLElement;
-  expect(tippy.style.visibility).toEqual('hidden');
-
   rows = container.querySelectorAll('.iui-table-body .iui-row');
```
and here too?

veekeys commented on packages/iTwinUI-react/src/core/Tooltip/Tooltip.test.tsx:
```diff
@@ -22,7 +22,7 @@ it('should toggle the visibility of tooltip on hover', () => {
   fireEvent.mouseLeave(getByText('Hover Here'));
   jest.runAllTimers();
-  expect(queryByText('some text')).not.toBeVisible();
+  expect(queryByText('some text')).not.toBeInTheDocument();
```
this is interesting...

veekeys replied:
![image](https://user-images.githubusercontent.com/81580355/162416151-da5be534-80ad-4a2b-b538-ad54bd599b90.png)

Why we didnt get it before? We need to carefully review our tests using `toBeVisible` cause we might get false results now :/
Or tippy behavior now somehow changed.. dont really understand

mayank99 replied:
I think tippy first becomes invisible then leaves the document. I'm not sure what the best way to test would be. My goal was to just fix all those failures in 10 different places, and I didn't find one single way to work everywhere.

veekeys commented on packages/iTwinUI-react/tsconfig.json:
```diff
@@ -19,6 +19,7 @@
   },
   "exclude": [
-    "lib/**/*"
+    "cjd/**/*",
```
cjd?

mayank99 replied:
fixed typo

veekeys reviewed (commented):
Storybook and playground did not load for me, so could not really check much.

veekeys commented on package.json:
```diff
@@ -1,145 +1,26 @@
-    "classnames": "^2.2.6",
-    "react-table": "^7.1.0",
-    "react-transition-group": "^4.1.1"
+    "build": "turbo run build",
```
well ok.. idk anymore..
I ran storybook from react package, seemed ok.
Then i ran again `yarn dev` and now it gave no errors :/ I still have all those prettier changes too, so not sure what happened here

mayank99 replied:
`yarn dev` should be fixed now.

Those errors were because the `@itwin/itwinui-react` package needs to be built before it can be used in the vite playground. So I added `^build` dependency and removed the `--parallel` flag.

Storybook should not be affected, I think it might have been because you had some other storybook running somewhere else.

mayank99 commented on .gitignore:
```diff
@@ -1,40 +1,31 @@
-node_modules/
-node_modules/*
+node_modules
+.pnp
```
I think these came from turbo create

mayank99 replied:
I've cleaned up the main gitignore

mayank99 commented on package.json:
```diff
@@ -1,145 +1,26 @@
-    "react",
-    "ui",
-    "ux"
+  "name": "turborepo-basic-shared",
```
Well, there's a lot of new dependencies so it makes sense idk

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Fix typo in tsconfig

mayank99 committed:
Clean up gitignore

mayank99 committed:
Whoops bad merge

mayank99 committed:
Remove leftovers

mayank99 committed:
Fix parallel yarn dev

mayank99 committed:
Fix `format` command

mayank99 commented on package.json:
```diff
@@ -1,145 +1,26 @@
+    "build": "turbo run build",
+    "dev": "turbo run dev --parallel",
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\"",
```
I've fixed the `format` command

veekeys committed:
tests

mayank99 committed:
Clean up nested gitignores

mayank99 committed:
Remove favicon and font family from playground

mayank99 committed:
Move vscode to root and ignore md from prettier

mayank99 committed:
Only keep README in root

mayank99 committed:
Update deps

mayank99 committed:
Even cleaner README

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Copy README on build

mayank99 committed:
Revert README

mayank99 committed:
Revert delay in Header.stories.tsx

veekeys commented on packages/iTwinUI-react/.eslintignore:
```diff
@@ -0,0 +1,13 @@
+node_modules
+public
+styles
+.next
```
so `json`, `public`, `styles`, `.next`, `coverage`, `dist` are all new and not sure if any of those are needed or even existing. What was the reason you have added `json`?

mayank99 replied:
i think it was trying to lint `turbo.json`. 

But I will need to look into this, as linting does not seem to work from command line

mayank99 replied:
I've cleaned up eslintignore so some of those (including `json`) have been removed

veekeys commented on packages/iTwinUI-react/.eslintrc.js:
```diff
@@ -0,0 +1,47 @@
+    'plugin:@typescript-eslint/recommended', // Uses the recommended rules from @typescript-eslint/eslint-plugin
+    'plugin:react-hooks/recommended',
+    'plugin:storybook/recommended',
+    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
```
`prettier` plugin is missing from here. Why?

mayank99 replied:
Added back `prettier` and removed `plugin:prettier`. See https://github.com/iTwin/iTwinUI-react/pull/613#issuecomment-1101715924

veekeys commented on .vscode/settings.json:
```diff
@@ -2,17 +2,9 @@
-    "**/lib/**": true,
-    "**/src/**/*.css": true
+  "[markdown]": {
+    "editor.defaultFormatter": "vscode.markdown-language-features"
```
sure about the removed props? None of them are needed anymore?

mayank99 replied:
i didnt see any difference

mayank99 replied:
I've made prettier as default formatter for all extensions except markdown. Seems to work fine

veekeys commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+    "@itwin/itwinui-css": "^0.53.3",
+    "@itwin/itwinui-icons-react": "^1.5.0",
+    "@itwin/itwinui-illustrations-react": "^1.0.1",
+    "@tippyjs/react": "^4.2.5",
```
we have `4.2.6` currently in `package.json`

bentleyvk replied:
Do we? https://github.com/iTwin/iTwinUI-react/blob/main/package.json#L46

veekeys replied:
oh...
not sure where I have looked then :D

veekeys commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+    "creevey": "^0.8.0-beta.0",
+    "eslint": "^8.12.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-node": "^11.1.0",
```
this is new

veekeys commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+    "react-dom": "^17.0.2",
+    "rimraf": "^2.6.2",
+    "storybook-dark-mode": "^1.0.9",
+    "tippy.js": "^6.3.1",
```
have `6.3.7` now

bentleyvk commented on turbo.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build"],
```
Can it depend on itself?

mayank99 replied:
The `^` means its dependencies' `build` needs to be completed first
https://turborepo.org/docs/reference/configuration#dependson

bentleyvk commented on .github/workflows/build.yml:
```diff
@@ -37,21 +37,21 @@ jobs:
       - run: yarn install --frozen-lockfile
 
-      - run: yarn test --coverage
+      - run: yarn workspace @itwin/itwinui-react test --coverage
```
Maybe we should move commands like these to the root package.json. What I mean: `"test": "yarn workspace @itwin/itwinui-react test"`

mayank99 replied:
we already have `test` in root package.json for running `test` in all repos

bentleyvk replied:
So why not use it?

mayank99 replied:
it doesn't accept `--coverage` flag

bentleyvk commented on apps/playground/src/main.tsx:
```diff
@@ -0,0 +1,64 @@
+import App from './App';
+
+const Shell = () => {
+  const [theme, setTheme] = React.useState<'light' | 'dark'>('dark');
```
Why not "os"?

gretanausedaite replied:
Why default dark?

mayank99 replied:
The `dark` was just personal preference when I made this playground. I've replaced it with a media query, which is easier to use than `os` for switching themes

bentleyvk commented on apps/playground/vite.config.ts:
```diff
@@ -0,0 +1,12 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import linaria from 'vite-plugin-linaria'
```
WHat it does?

mayank99 replied:
it's the CSS-in-JS lib I'm using

bentleyvk commented on apps/playground/vite.config.ts:
```diff
@@ -0,0 +1,12 @@
+import { defineConfig } from 'vite'
```
Why it is not formated? I see `;` is missing.

bentleyvk commented on packages/iTwinUI-react/.eslintrc.js:
```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
```
Why this file was changed at all? Also quetes changed too.

mayank99 replied:
The contents of this file are now in `eslint-preset.js`

bentleyvk commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+  "scripts": {
+    "build": "yarn clean && yarn copy-files && tsc -p tsconfig.cjs.json && tsc -p tsconfig.esm.json",
+    "build:watch": "yarn clean && yarn copy-files && concurrently \"tsc -p tsconfig.cjs.json --watch\" \"tsc -p tsconfig.esm.json --watch\"",
+    "clean": "yarn clean:coverage && rimraf esm && rimraf cjs",
```
Will `rimraf` work on all platforms?

mayank99 replied:
`rimraf` is the name of the package we use for cross-platform `rm` command. It works on macOS and it was working on Windows too (It's the same as before)

bentleyvk commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "createComponent": "node scripts/createComponent.js",
+    "format": "prettier --config .prettierrc **/*.{tsx,ts,js} --ignore-path .gitignore --write",
```
Why using `.gitignore` for ignore paths? We should mix config files.

mayank99 replied:
everything that is gitignored should also be prettierignored, so it makes sense to me

bentleyvk commented on package.json:
```diff
@@ -1,145 +1,25 @@
-    "ts-loader": "^8.0.1",
-    "ts-node": "^8.0.2",
-    "typescript": "^4.4.3",
+    "turbo": "latest",
```
Is it ok to specify version like this?

mayank99 replied:
this came from turbo's `npx create`. Want me to change it to `1.x` or something?

mayank99 replied:
Locked to 1.2.4

bentleyvk commented on package.json:
```diff
@@ -1,145 +1,25 @@
-    "ts-node": "^8.0.2",
-    "typescript": "^4.4.3",
+    "turbo": "latest",
     "webpack": "5"
```
Why it is still needed?

mayank99 replied:
It's in the PR description
<img width="774" alt="image" src="https://user-images.githubusercontent.com/9084735/162758270-6bf5540b-b928-4ab8-bacc-2cb4633f1af4.png">

bentleyvk commented on package.json:
```diff
@@ -1,145 +1,25 @@
-    "**/*.scss",
-    "**/*.css"
-  ]
+  "packageManager": "yarn@1.22.18"
```
Have you checked what is a prefered package manager? Maybe we should switch to npm or pnpm.

mayank99 replied:
i think pnpm is the preferred package manager, but i kept yarn because that's what we were using before

bentleyvk replied:
Could you investigate main differences between pnpm and yarn because more and more I am seeing people saying to not 
 to use yarn

mayank99 replied:
Maybe not in this PR? I would need to update all the documentation and workflows everywhere.

It's just a package manager so it could easily be done later. There is even a command to migrate lockfile: https://pnpm.io/cli/import

bentleyvk commented on .eslintrc.js:
```diff
@@ -1,44 +0,0 @@
-/*---------------------------------------------------------------------------------------------
```
Why it is bad to have eslint settings in the root?

mayank99 replied:
Moved configs in shared package

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Use `yarn dev` in github workflow

mayank99 committed:
Merge branch 'main' into mayank/monorepo

gretanausedaite reviewed (commented):
Repo starts, unit test runs and passes. 
Main concerns: 
1. Why `yarn dev` starts both playground and storybook? Is there a reason why it's more useful to look at both when coding/creating than just playground or storybook?
2. Visual tests are missing (at least I didn't find how to run them). Also creevey as is won't run on Mac M1, we could use this new repo opportunity to switch to other testing suite or fix creevey.

mayank99 commented:
> 1. Why `yarn dev` starts both playground and storybook? Is there a reason why it's more useful to look at both when coding/creating than just playground or storybook?

That's the default behavior of turborepo and it also makes sense to me. When I'm working on any component, I have both storybook and test app running.

When we add website to this repo, it will also start with the same command

> 2. Visual tests are missing (at least I didn't find how to run them). Also creevey as is won't run on Mac M1, we could use this new repo opportunity to switch to other testing suite or fix creevey.

What visual tests are missing?

<img width="378" alt="image" src="https://user-images.githubusercontent.com/9084735/163392703-55f8cb7a-cf70-4e0b-96ef-038cc9f8ee1d.png">

Good point about using this opportunity. I don't think we will be able to fix creevey but we could switch to cypress or something.

mayank99 committed:
Fix linting and move configs into separate package

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Match os theme in playground

mayank99 committed:
Disallow pushing with eslint warnings

mayank99 committed:
Clean up eslintignore

mayank99 committed:
Fix bad merge

mayank99 commented:
I've fixed linting/formatting issues by making the following changes:
- As per [turborepo's recommendation](https://github.com/vercel/turborepo/tree/main/packages/create-turbo/templates/_shared_ts/packages/config), I moved all configs into a shared `configs` package and added that package as a dev dependency in other packages.
- I was facing conflicts between prettier and eslint, so I looked at their [docs](https://prettier.io/docs/en/integrating-with-linters.html) and they recommend using _only_ `eslint-config-prettier`, so I've removed `eslint-plugin-prettier`.
- I was facing problems running `yarn lint` so I looked at their docs and they [recommend adding quotes around paths](https://eslint.org/docs/user-guide/command-line-interface#:~:text=If%20you%20want%20to%20use%20node%20glob%20syntax%2C%20you%20have%20to%20quote%20your%20parameter), which seemed to fixed the problem.
- As per [lint-staged recommendation](https://github.com/okonet/lint-staged#how-to-use-lint-staged-in-a-multi-package-monorepo), I've added `lint-staged` command to root package.json to forward it to individual packages. Tested and it's working fine after I added `--max-warnings=0` to the eslint command.

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Merge branch 'main' into mayank/monorepo

veekeys commented on packages/configs/eslint-preset.js:
```diff
@@ -0,0 +1,46 @@
+    'plugin:@typescript-eslint/recommended', // Uses the recommended rules from @typescript-eslint/eslint-plugin
+    'plugin:react-hooks/recommended',
+    'plugin:storybook/recommended',
+    'prettier',
```
we had this before too `plugin:prettier/recommended`

mayank99 replied:
<img width="789" alt="image" src="https://user-images.githubusercontent.com/9084735/164473694-3686c816-ab41-4987-8cfe-a8e53bcb4ff6.png">

veekeys commented on packages/configs/eslint-preset.js:
```diff
@@ -0,0 +1,46 @@
+  ],
+  rules: {
+    // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+    'react/prop-types': 'off',
```
and here this `"prettier/prettier": [
      1
    ],`

mayank99 replied:
<img width="789" alt="image" src="https://user-images.githubusercontent.com/9084735/164474122-e2a8e93b-6314-45e5-8ebb-b733f3c4fe29.png">

veekeys commented on packages/iTwinUI-react/tsconfig.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "configs/tsconfig.base.json",
```
is this path correct? it shows error to me

mayank99 replied:
It is correct. `configs` is the name of the package where I've moved this file.

https://user-images.githubusercontent.com/9084735/164474724-79c0a9fd-5dd4-42f5-ae27-68e16c623fff.mov

veekeys commented:
Linter seems to work fine. I have tried messing up code and checking if on save fixes minor things.
Also checked, if yarn lint will catch warning (like `any` type), which worked fine too.
Really appreciate the links for recommendations, cause otherwise, it would be hard to understand the structure.
Just weird, that tsconfig error was not notified anywhere.. wonder if there are some more like that left

mayank99 committed:
Merge branch 'main' into mayank/monorepo

mayank99 committed:
Wrap tooltip timer in `act` to fix warning

mayank99 committed:
Lock to patch version of `turbo`

mayank99 commented on package.json:
```diff
@@ -1,145 +1,26 @@
+    "dev": "turbo run dev --parallel",
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\"",
+    "test": "turbo run test"
```
Fixed in [`80daaea`](https://github.com/iTwin/iTwinUI-react/pull/613/commits/80daaea52a96e4f121b7054f32b8fca4ac660ccb)

mayank99 committed:
Remove `eslint-plugin-node`

veekeys reviewed (approved):
Good job!
Welcome, Turborepo!

mayank99 committed:
Include md files in package

mayank99 commented on packages/iTwinUI-react/package.json:
```diff
@@ -0,0 +1,147 @@
+    "lint": "eslint --ext .js,.ts,.tsx --max-warnings=0",
+    "lint:fix": "yarn lint --fix && yarn lint:copyright --fix */**/*.{js,ts,tsx}",
+    "lint:copyright": "node scripts/copyrightLinter.js",
+    "copy-files": "cpx ../../README.md .",
```
Ok turns out they were not included.

Fixed in [`0b92fe3`](https://github.com/iTwin/iTwinUI-react/pull/613/commits/0b92fe31f3682462c887a68fcccafded30a111f1) and tested using `npm pack`

![image](https://user-images.githubusercontent.com/9084735/164534091-0f883759-d2fa-410e-9abb-349c157397e2.png)

mayank99 committed:
Fix `copy-files` syntax

bentleyvk commented on .github/workflows/release-please.yml:
```diff
@@ -28,17 +28,17 @@ jobs:
         with:
-          node-version: 12.x
+          node-version: 16.x
           registry-url: https://registry.npmjs.org/
```
No need for cache here?

mayank99 replied:
it hasn't changed from before.

iirc i didn't use cache because we should take fresh deps for release

bentleyvk reviewed (approved):
Monorepo it is
